api/src/main/java/org/apache/iceberg/transforms/TimestampTransform.java

```diff
@@ -127,6 +152,57 @@ public String toHumanString(Integer value) {
 
   @Override
   public String toString() {
-    return name;
+    if (zoneOffset.getTotalSeconds() == 0) {
+      return name;
+    } else {
+      return name + "[" + zoneOffset.getId() + "]";
+    }
+  }
+
+  @Override
+  public boolean equals(Object other) {
+    if (this == other) {
+      return true;
+    }
+    if (other == null || getClass() != other.getClass()) {
+      return false;
+    }
+    TimestampTransform that = (TimestampTransform) other;
+    return granularity == that.granularity &&
+        Objects.equal(name, that.name) &&
+        Objects.equal(zoneOffset, that.zoneOffset);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hashCode(granularity, name, zoneOffset);
+  }
+
+  public ChronoUnit getGranularity() {
```

Comment:
In Iceberg, we don't use `get` because it doesn't add value. Can you change this to `granularity`?